File: apps/ios/Barns/BarnsApp/Domain/UseCases/Catalog/GetCareGuidesUseCase.swift
/// Returns official care guides. With no ids, returns all guides; with ids,
/// returns the matching guides (for a catalog item's `careGuideIds`).
struct GetCareGuidesUseCase {
    private let repository: CareGuideRepository

    init(repository: CareGuideRepository) {
        self.repository = repository
    }

    func execute() async throws -> [CareGuide] {
        try await repository.careGuides()
    }

    func execute(ids: [CareGuide.ID]) async throws -> [CareGuide] {
        try await repository.careGuides(ids: ids)
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/catalog/GetCareGuidesUseCase.kt
package com.barns.app.domain.usecase.catalog

import com.barns.app.domain.model.CareGuide
import com.barns.app.domain.repository.CareGuideRepository

/**
 * Returns official care guides. With no ids, returns all guides; with ids,
 * returns the matching guides (for a catalog item's `careGuideIds`).
 */
class GetCareGuidesUseCase(
    private val repository: CareGuideRepository,
) {
    suspend fun execute(): List<CareGuide> = repository.careGuides()

    suspend fun execute(ids: List<String>): List<CareGuide> = repository.careGuides(ids)
}
